File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Models/Cuenta.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BanquitoServer_Soap_DotNet_GR01.Models
{
    /// <summary>
    /// Entidad Cuenta bancaria
    /// </summary>
    [Table("Cuenta")]
    public class Cuenta
    {
        [Key]
        [Column("CuentaId")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long CuentaId { get; set; }

        [Required]
        [Column("NumeroCuenta")]
        [MaxLength(20)]
        [Index("IX_Cuenta_NumeroCuenta", IsUnique = true)]
        public string NumeroCuenta { get; set; }

        [Required]
        [Column("TipoCuenta")]
        [MaxLength(20)]
        public string TipoCuenta { get; set; }

        [Required]
        [Column("Saldo")]
        public decimal Saldo { get; set; }

        [Required]
        [Column("FechaApertura")]
        public DateTime FechaApertura { get; set; }

        [Required]
        [Column("Estado")]
        [MaxLength(20)]
        public string Estado { get; set; }

        // Foreign Key
        [Required]
        [Column("ClienteId")]
        [ForeignKey("Cliente")]
        [Index("IX_Cuenta_ClienteId")]
        public long ClienteId { get; set; }

        // Propiedades de navegación
        public virtual Cliente Cliente { get; set; }
        public virtual ICollection<Movimiento> Movimientos { get; set; }

        public Cuenta()
        {
            Movimientos = new HashSet<Movimiento>();
            Estado = "ACTIVA";
            Saldo = 0;
        }
    }
}

File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. SERVIDOR/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Models/Cuenta.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.util.HashSet;
import java.util.Set;

/**
 * Entidad Cuenta bancaria
 */
@Entity
@Table(name = "Cuenta")
@Data
@NoArgsConstructor
public class Cuenta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "CuentaId")
    private Long cuentaId;

    @Column(name = "NumeroCuenta", nullable = false, length = 20, unique = true)
    private String numeroCuenta;

    @Column(name = "Saldo", nullable = false, precision = 12, scale = 2)
    private BigDecimal saldo;

    @Column(name = "TipoCuenta", length = 20)
    private String tipoCuenta;

    @Column(name = "Estado", length = 20)
    private String estado;

    // Foreign Key
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ClienteId", nullable = false)
    private Cliente cliente;

    // Relaciones
    @OneToMany(mappedBy = "cuenta", fetch = FetchType.LAZY)
    private Set<Movimiento> movimientos = new HashSet<>();
}
